src/platform/linux.rs

```diff
@@ -271,11 +307,42 @@ impl Platform for PlatformImpl {
     }
 
     fn mounts(&self) -> io::Result<Vec<Filesystem>> {
-        Err(io::Error::new(io::ErrorKind::Other, "Not supported"))
+        read_file("/proc/mounts")
+            .and_then(|data| {
+                proc_mounts(data.as_bytes()).to_result().map_err(|err| {
+                    io::Error::new(io::ErrorKind::InvalidData, err)
+                })
+            })
+            .map(|mounts| {
+                mounts
+                    .into_iter()
+                    .filter_map(|mount| {
+                        Statvfs::for_path(&mount.target as &str)
+                            .map(move |s| {
+                                Filesystem {
+                                    files: s.f_files as usize,
+                                    free: ByteSize::b(s.f_bfree as usize * s.f_bsize as usize),
+                                    avail: ByteSize::b(s.f_bavail as usize * s.f_bsize as usize),
+                                    total: ByteSize::b(s.f_blocks as usize * s.f_bsize as usize),
+                                    name_max: s.f_namemax as usize,
+                                    fs_type: mount.fstype,
+                                    fs_mounted_from: mount.source,
+                                    fs_mounted_on: mount.target,
+                                }
+                            })
+                            .ok()
+                    })
+                    .collect()
+            })
     }
 
     fn mount_at<P: AsRef<path::Path>>(&self, path: P) -> io::Result<Filesystem> {
-        Err(io::Error::new(io::ErrorKind::Other, "Not supported"))
+        self.mounts().and_then(|mounts| {
+            mounts
+                .into_iter()
+                .find(|mount| Path::new(&mount.fs_mounted_on) == path.as_ref())
```

Comment:
I think here we're executing the syscall on all the mounts, only to reject the ones that don't match. Which is a bit of a shame because the filter just needs `fs_mounted_on`, which doesn't require the syscall.